src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,220 @@
 package internet.shop.dao.impl;
 
 import internet.shop.dao.CarDao;
-import internet.shop.db.Storage;
+import internet.shop.exception.DataProcessingException;
 import internet.shop.lib.Dao;
 import internet.shop.model.Car;
+import internet.shop.model.Driver;
+import internet.shop.model.Manufacturer;
+import internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Optional;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 @Dao
 public class CarDaoImpl implements CarDao {
     @Override
     public Car create(Car car) {
-        Storage.addCar(car);
+        String queryForCar = "INSERT INTO cars (manufacturer_id, model) values (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatementOfCar = connection.prepareStatement(queryForCar,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatementOfCar.setLong(1, car.getManufacturer().getId());
+            preparedStatementOfCar.setString(2, car.getModel());
+            preparedStatementOfCar.executeUpdate();
+
+            ResultSet generatedKeys = preparedStatementOfCar.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't create the data - " + car, exception);
+        }
         return car;
     }
 
     @Override
     public Optional<Car> get(Long id) {
```

Comment:
here, as well as in getAll() and getAllByDriver() methods we should use 2 queries - first will fetch data about the car and its manufacturer

after the first query, call the separate query that will return all drivers data for this car (do it in a separate private method)

let's not use ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY for now